tests/transports-integrations/tests/utils/common.py

```
@@ -0,0 +1,416 @@
+        return False
+
+    response_dict = response.__dict__ if hasattr(response, "__dict__") else response
+
+    for field in expected_fields:
+        if field not in response_dict:
+            return False
+
+    return True
+
+
+def extract_tool_calls(response: Any) -> List[Dict[str, Any]]:
```

Ruff (0.11.9)

`extract_tool_calls` is too complex (11 > 10)

(C901)